pandas/core/strings.py

```diff
@@ -311,8 +311,12 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     Parameters
     ----------
-    pat : string
-        Character sequence or regular expression
+    pat : string or compiled regex
```

Comment:
the signature needs to change for ``case=None`` and ``flags=None`` (and then set them if not a is_compiled_re (and they are None)